scripts/lib-franklin.js

```diff
@@ -534,6 +703,17 @@ export function getIconTypebyPath(url) {
 export function decorateButtons(element, options = {}) {
   const mergedOptions = { ...{ decorateClasses: true, excludeIcons: ['internal'] }, ...options };
   element.querySelectorAll('a').forEach((a) => {
+    const videoType = getVideoType(a.href);
+    if (['youtube', 'mp4'].includes(videoType)) {
+      const videoModal = buildVideoModal(a.href, videoType);
+      const videoClose = videoModal.querySelector('button.video-modal-close');
+      videoClose.addEventListener('click', () => toggleVideoOverlay(videoModal));
+      a.addEventListener('click', (e) => {
+        e.preventDefault();
+        toggleVideoOverlay(videoModal);
+      });
+      element.appendChild(videoModal);
+    }
```

Comment:
Let's add comments and format this similar to the suggestion above.

Reply:
@proeung for the first point, could you please explain a bit on that overlay issue?